modules/standard/BigInteger.chpl

```diff
@@ -2767,88 +2825,95 @@ module BigInteger {
+      mpz_nextprime(result.mpz, a.mpz);
     } else {
-      const thisLoc = chpl_buildLocaleID(this.localeId, c_sublocid_any);
-
-      on __primitive("chpl_on_locale_num", thisLoc) {
-        var a_ = a;
-
-        mpz_nextprime(this.mpz, a_.mpz);
+      const resultLoc = chpl_buildLocaleID(result.localeId, c_sublocid_any);
+      on __primitive("chpl_on_locale_num", resultLoc) {
+        const a_ = a.localize();
+        mpz_nextprime(result.mpz, a_.mpz);
       }
     }
   }
 
+  @deprecated(notes="bigint.nextprime method is deprecated - please use the standalone function :proc:`~BigInteger.nextprime`")
+  proc bigint.nextprime(const ref a: bigint) {
+    BigInteger.nextprime(this, a);
+  }
 
-
-  // gcd
-  proc bigint.gcd(const ref a: bigint, const ref b: bigint) {
+  proc gcd(ref result: bigint, const ref a: bigint, const ref b: bigint) {
     if _local {
-      mpz_gcd(this.mpz, a.mpz, b.mpz);
-
-    } else if this.localeId == chpl_nodeID &&
+      mpz_gcd(result.mpz, a.mpz, b.mpz);
+    } else if result.localeId == chpl_nodeID &&
               a.localeId    == chpl_nodeID &&
               b.localeId    == chpl_nodeID {
-      mpz_gcd(this.mpz, a.mpz, b.mpz);
-
+      mpz_gcd(result.mpz, a.mpz, b.mpz);
     } else {
-      const thisLoc = chpl_buildLocaleID(this.localeId, c_sublocid_any);
-
-      on __primitive("chpl_on_locale_num", thisLoc) {
-        var a_ = a;
-        var b_ = b;
-
-        mpz_gcd(this.mpz, a_.mpz, b_.mpz);
+      const resultLoc = chpl_buildLocaleID(result.localeId, c_sublocid_any);
+      on __primitive("chpl_on_locale_num", resultLoc) {
+        const a_ = a.localize();
+        const b_ = b.localize();
+        mpz_gcd(result.mpz, a_.mpz, b_.mpz);
       }
     }
   }
 
-  proc bigint.gcd(const ref a: bigint, b: int) {
+  @deprecated(notes="bigint.gcd method is deprecated - please use the standalone function :proc:`~BigInteger.gcd`")
+  proc bigint.gcd(const ref a: bigint, const ref b: bigint) {
+    BigInteger.gcd(this, a, b);
+  }
+
+  proc gcd(ref result: bigint, const ref a: bigint, b: int) {
     if b >= 0 {
-      this.gcd(a, b : uint);
+      BigInteger.gcd(result, a, b : uint);
 
     } else {
-      this.gcd(a, new bigint(b));
+      BigInteger.gcd(result, a, new bigint(b));
     }
   }
 
-  proc bigint.gcd(const ref a: bigint, b: uint) {
-    const b_ = b.safeCast(c_ulong);
+  @deprecated(notes="bigint.gcd method is deprecated - please use the standalone function :proc:`~BigInteger.gcd`")
+  proc bigint.gcd(const ref a: bigint, b: int) {
+    BigInteger.gcd(this, a, b);
+  }
 
+  proc gcd(ref result: bigint, const ref a: bigint, b: uint) {
+    const b_ = b.safeCast(c_ulong);
     if _local {
-      mpz_gcd_ui(this.mpz, a.mpz, b_);
-
-    } else if this.localeId == chpl_nodeID &&
+      mpz_gcd_ui(result.mpz, a.mpz, b_);
+    } else if result.localeId == chpl_nodeID &&
               a.localeId    == chpl_nodeID {
-      mpz_gcd_ui(this.mpz, a.mpz, b_);
-
+      mpz_gcd_ui(result.mpz, a.mpz, b_);
```

Comment:
Similar comment; if we could switch all of these multi-pronged checks to localize everything except for `result`, that would be good.

Could be worth splitting it into a separate PR so that this PR doesn't continue to grow and grow! (or you can just open an issue if you can't get to it now and we can get to it in time)